pkg/services/sqlstore/searchstore/filters_test.go

```diff
@@ -57,3 +57,55 @@ func TestFolderUIDFilter(t *testing.T) {
 		})
 	}
 }
+
+func TestFolderUIDFilterWithNestedFoldersEnabled(t *testing.T) {
```

Comment:
We could add these test cases to the test above. The way to do so would be to expand the `testCases` struct to track whether `NestedFoldersEnabled` should be set to `true` or `false`. This new field in `testCases` could also be called "nestedFoldersEnabled".

So instead of setting `NestedFoldersEnabled` to `true` on line 102, we would set it to  `tc.nestedFoldersEnabled`. The test cases above would have `tc.nestedFoldersEnabled` set to `false` while the new ones you added would have it set to `true`.